src/fork.rs

```diff
@@ -458,28 +459,46 @@ impl ForkDetails {
         miniblock: u64,
         chain_id: Option<L2ChainId>,
         cache_config: CacheConfig,
-    ) -> Self {
+    ) -> eyre::Result<Self> {
         let url = network.to_url();
-        let block_details = client
+        let opt_block_details = match client
             .get_block_details(L2BlockNumber(miniblock as u32))
             .await
-            .unwrap()
-            .unwrap_or_else(|| panic!("Could not find block {:?} in {:?}", miniblock, url));
-
-        let root_hash = block_details
-            .base
-            .root_hash
-            .unwrap_or_else(|| panic!("fork block #{} missing root hash", miniblock));
-        let block = client
-            .get_block_by_hash(root_hash, true)
-            .await
-            .expect("failed retrieving block")
-            .unwrap_or_else(|| {
-                panic!(
+        {
+            Ok(obd) => obd,
+            Err(error) => {
+                return Err(eyre!(error));
+            }
+        };
+        let block_details = match opt_block_details {
```

Comment:
there is more ergonomic alternative for such matches:
```
opt_block_details .ok_or_else(|| {
    eyre!("Could not find block {:?} in {:?}", miniblock, url)
})?;
```

I think it worth to switch to it.

Reply:
OK, switched.